### PR TITLE
Add support for kustomize

### LINKED ISSUE #370: Support Kustomize
Kustomize has merged into mainline. We are going to need to ship some sort of support of it for the next version of Kubernetes.

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
Add a new kustomize.Directory class to each SDK that allows
Pulumi to directly manage kustomizations.

TODO:

- [X] NodeJS
- [x] Python
- [x] C#
- [x] Go

### Related issues (optional)
Fix #370

### CODE REVIEW

File: provider/pkg/gen/dotnet-templates/Kustomize/Directory.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2016-2020, Pulumi Corporation
```
Comment (metral) [inline provider/pkg/gen/dotnet-templates/Kustomize/Directory.cs:1]: It doesn't appear this has an auto-generated warning like the other languages do. Or was this manually written?

Comment (lblackstone) [reply provider/pkg/gen/dotnet-templates/Kustomize/Directory.cs:1]: Will add

File: provider/pkg/provider/invoke_kustomize.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		var err error
+
+		// Create a temp dir.
+		var temp string
+		if temp, err = ioutil.TempDir("", "kustomize-"); err != nil {
+			return nil, errors.Wrap(err, "failed to create temp directory for remote kustomize directory")
+		}
+		defer os.RemoveAll(temp)
+
+		path, err = workspace.RetrieveGitFolder(directory, temp)
```
[the remark this answers is not in the document]
Comment (lblackstone) [reply provider/pkg/provider/invoke_kustomize.go:44]: The package I'm using for git support doesn't expose auth options. I'm inclined to leave it as is for now and add support later if it's needed.

File: provider/pkg/provider/util.go [hunk 3]
```diff
@@ -11,6 +11,10 @@ import (
 	clientapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
+func isUrl(path string) bool {
+	return strings.HasPrefix(path, "http")
```
[the remark this answers is not in the document]
Comment (lblackstone) [reply provider/pkg/provider/util.go:15]: Updated the logic to check if the directory exists locally, and fallback to trying as a git repo if not.

**Pull request merged**